python/src/iceberg/expressions/base.py

```diff
@@ -278,6 +278,9 @@ def __str__(self):
     def __repr__(self):
         return f"Accessor(position={self._position})"
 
+    def __eq__(self, other):
+        return other and self._position == other.position
```

Comment:
We don't manipulate the `self._position` property so this is just a nitpick, but should we make this `self.position` just so it goes through that path?